tests/test_sparkkernelbase.py

```diff
@@ -65,16 +65,21 @@ def test_get_config():
     conf.load()
 
 
+# We check that the kernel goes into error state when configuration
+# is not set properly
 @with_setup(_setup, _teardown)
 def test_get_config_not_set():
     conf.override_all({})
-    try:
-        kernel._get_configuration()
+    kernel._get_configuration()
+    assert kernel._fatal_error
+    conf.load()
 
-        # Above should have thrown because env var not set
-        assert False
-    except ValueError:
-        assert send_error_mock.call_count == 1
```

Comment:
That's not really possible in the way that we were doing it before.

Actually, what the way it was implemented before *should have* worked, except if you throw an error in the `__init__` method of a kernel, Jupyter assumes that the kernel has crashed.  So displaying an error to the user before they've even entered any input is not generally possible, I think.

The idea behind this change is that now the `__init__` method will mark the kernel as being in an error state when it starts up so that the error can be thrown when the user tries to run a cell.  This unit test is testing that.